### PR TITLE
Markdown & LaTeX rendering, code blocks w/ syntax highlighting, styled tables, headings, and lists.

### LINKED ISSUE #94: Render Markdown and LaTeX in chat responses
Render Markdown and LaTeX in the chat interface to improve usability, support lists, headers, equations, etc.

Screenshot 2024-03-12 at 6 37 15 PM

### PR DESCRIPTION
**Pull request by kaeladair**
Added Markdown & LaTeX rendering using Katex, styled tables, lists, and headers, added code blocks with syntax highlighting.

Fixes #94 

Had to add to main chat prompt to ensure proper delimiters and formatting from gpt-4.

Styling is in globals.css for now, as I am not super familiar with tailwind. Easy/quick fix if it is a problem.

Screenshot 2024-03-13 at 5 44 51 PM
Screenshot 2024-03-13 at 5 45 43 PM
Screenshot 2024-03-13 at 5 45 57 PM

### CODE REVIEW

File: app/src/styles/globals.css [hunk 1]
```diff
@@ -58,3 +58,118 @@
...
+  overflow-x: auto; /* Ensures horizontal scrolling for long lines of code */
+  font-family: 'Fira Code', 'Consolas', 'Monaco', monospace; /* Modern monospace font */
+  font-size: 0.95em; /* Slightly smaller font size for a more compact look */
+  line-height: 1.4; /* Improved line spacing for readability */
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2); /* Subtle shadow for depth */
+  margin-top: 10px;
+  margin-bottom: 10px;
+}
+
+/* Styling for headers, text, lists */
```
[the remark this answers is not in the document]
Comment (kaeladair) [reply app/src/styles/globals.css:92]: Will do

**Comment by adamcohenhillel** [comment]
Pretty awesome stuff - will merge after resolving @Jacksonmills comments!

**Review by adamcohenhillel — APPROVED** [review]

File: app/src/styles/globals.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaeladair) [reply app/src/styles/globals.css]: Should be better now, header/list styles will only apply to the chatlog.

[the remark this answers is not in the document]
Comment (Jacksonmills) [reply app/src/styles/globals.css]: closer! let me make some changes to show you how to do it all in tailwind

**Comment by Jacksonmills** [comment]
@kaeladair https://github.com/kaeladair/ADeus/pull/2

I have a PR open on your fork that should clean up everything, however when I ask for math rendering it doesn't give it to me despite me not touching those globals

**Review by Jacksonmills — CHANGES_REQUESTED** [review]
I put in some changes as a PR on your fork, let me know what you think there, one bug with the math stuff or I just suck at prompting

**Comment by kaeladair** [comment]
> I put in some changes as a PR on your fork, let me know what you think there, one bug with the math stuff or I just suck at prompting

Looks great! Trying to figure out the bug and will get back to you soon

**Comment by Jacksonmills** [comment]
> > I put in some changes as a PR on your fork, let me know what you think there, one bug with the math stuff or I just suck at prompting
> 
> Looks great! Trying to figure out the bug and will get back to you soon

yeah I saw your screenshot of the equations so I tried asking for one but no luck

**Comment by kaeladair** [comment]
> > > I put in some changes as a PR on your fork, let me know what you think there, one bug with the math stuff or I just suck at prompting
> > 
> > 
> > Looks great! Trying to figure out the bug and will get back to you soon
> 
> yeah I saw your screenshot of the equations so I tried asking for one but no luck

Everything is working on my end, with the bug being that the GPT is responding with some markdown headers mixed with LaTeX and is doesn't render properly. Will try to prompt it.

**Review by Jacksonmills — APPROVED** [review]

**Comment by adamcohenhillel** [comment]
merging!

**Pull request merged**